Add event tracking to website

eirizarry opened the pull request:
Second attempt to add event tracking to website. This has been successfully tested in dev. 

- Moved the ga event tracking code to page header of baseof.html. Keeping the ga code with the original tracking snippet still works, but google docs said it should be in a page header rather than in same file. 

- Web and Android examples were not being properly tracked because the event target was the img/header. Adds method to recursively find href in parent anchor element

- Excludes internal clicks (ga already has a filter to exclude localhost, so that will not be an issue)

eirizarry committed:
Fix ga event tracking

- Move event tracking snippet into page header
- Add and test onclick attribute to web links

eirizarry committed:
Add onclick attribute to site links

pdavidc reviewed (changes requested):
Explicitly modifying every link to support outbound tracking is going to be difficult to maintain. Let's try doing something at the DOM level, such as these suggestions:
https://stackoverflow.com/questions/8492344/javascript-attach-an-onclick-event-to-all-links

eirizarry committed:
Remove onclick attributes from site links

eirizarry committed:
Add event-tracking.js at DOM level

eirizarry commented:
I successfully added event tracking at the DOM level, but I have run into another problem. First, this tracking is also going to include internal clicks. In the example report below, you can see that everything I click on to go to another site page is recorded. This tracking is redundant because we can already see what pages people are viewing. 

![screen shot 2017-12-13 at 4 11 23 pm](https://user-images.githubusercontent.com/23271817/33970590-b1986f4a-e028-11e7-81a7-a3b6283caa27.png)

I am wondering if there is a way to exclude the internal clicks? Something like: 

     if (domainName includes worldwind.arc.nasa.gov) { 
          exclude 'click'
     }
     else {
          function handleOutboundLinkClicks()
     }

This also does not accurately track clicking on Web examples, or Android examples and tutorials--the label is reported as (not set)--see above. Even though the pages are rendered as html, and I am using getElementsByTagName, all these instances were written as Hugo shortcodes. 

@zglueck, do you have any guidance on this?

eirizarry committed:
Track only outbound links

- recursively find href for Web and Android examples
- exclude tracking internal clicks

eirizarry committed:
Reformat code

eirizarry commented:
@zglueck was able to help me solve the remaining issues. Everything has been successfully tested.

zglueck reviewed (changes requested):
I overlooked a simpler approach this afternoon. Take a look at the `event.currentTarget.href` property. It looks like it would eliminate the recursive search.

eirizarry committed:
Remove recursive method for event tracking

- Use event.currentTarget.href instead

eirizarry committed:
Add gh-site as an internal host

eirizarry committed:
Remove redundant code

pdavidc commented:
@zglueck Have you had a chance to review the current state of this pull request?

zglueck reviewed (approved):
Looks good!
